openlibrary/macros/FulltextSearchSuggestion.html

```diff
@@ -0,0 +1,34 @@
+$def with(query, results, page=1)
+
+$def render_snippet(query, doc):
+    $:macros.FulltextSuggestionSnippet(query, doc=doc)
+
+$if results and results.get('hits'):
+  $ hits = results['hits'].get('hits', [])[:4]
+  $ num_found = commify(results['hits'].get('total', 0))
+
+  $:macros.LoadingIndicator(_("Checking for Search Inside matches"))
+    <div id="fulltextSuggestions" class="fulltext-suggestions">
+      <div class="fulltext-suggestions__header">
+        <a class="fulltext-suggestions__icon" data-ol-link-track="SearchInsideSuggestion|SeeAllClick" href="/search/inside?$urlencode(dict(q=query))" title="$_('Search Inside Icon')">
+            <img src="/static/images/icons/icon_search-inside.svg"
+              width="30" height="30"
+              alt="$_('search inside icon')"/>
+        </a>
+        <h3 class="fulltext-suggestions__title">
+          <a data-ol-link-track="SearchInsideSuggestion|SeeAllClick"
+          href="/search/inside?$urlencode(dict(q=query))">$:_('Search Inside</a> found these books with matching passages...')
```

Comment:
I wonder if this should be...
```suggestion
          href="/search/inside?$urlencode(dict(q=query))">$_('Search Inside')</a> $_('found these books with matching passages...')
```

Reply:
Ahh yes a fun case! Typically we'd want to keep the whole sentence together, in case word order varies in the different languages, while escaping out the Python variable with `%()s` syntax, i.e.
```suggestion
          $:_('<a data-ol-link-track="SearchInsideSuggestion|SeeAllClick" href="/search/inside?%(query)s">Search Inside</a> found these books with matching passages...', query=urlencode(dict(q=query)))
```

I agree it's an annoying-looking link, but seems worthwhile to me for keeping the sentence together.

Reply:
@rebecca-shoptaw is there any challenge of a value getting passed in to `query` which then results in a js (or xss) exploit after `$:` renders the query as html?

Reply:
4 options:
1. use `websafe` for the query param -- `query=websafe(urlencode(dict(q=query)))
2. Move all the text "Search Inside found these..." into the `<a>...</a>`
3. Remove the `<a>` tag entirely :)

Actually 😆 we could internationalize both separately in two parts if we do this:
 
<img width="625" alt="Screenshot 2024-08-15 at 11 38 27 AM" src="https://github.com/user-attachments/assets/7daf6b33-1046-45c9-ab61-3ebae2cb6352">